index.js

```diff
@@ -68,8 +69,82 @@ async function registerCrud(fastify, { modelName, isView }) {
   fastify.decorate('jsonSchemaGenerator', model.jsonSchemaGenerator)
   fastify.decorate('jsonSchemaGeneratorWithNested', model.jsonSchemaGeneratorWithNested)
   fastify.decorate('modelName', modelName)
-  const prefix = model.definition.endpointBasePath
-  fastify.register(httpInterface, { prefix, isView })
+  fastify.register(httpInterface, { prefix, registerGetters: true, registerSetters: !isView })
+}
+
+async function registerViewCrud(fastify, { modelName, lookups }) {
+  if (!fastify.mongo) { throw new Error('`fastify.mongo` is undefined!') }
+  if (!modelName) { throw new Error('`modelName` is undefined!') }
+
+  fastify.log.trace({ modelName }, 'Registering View CRUD')
+
+  const { definition, viewDependencies, crudService } = fastify.models[modelName]
+  const prefix = definition.endpointBasePath
+
+  fastify.decorate('crudService', viewDependencies.crudService)
+  fastify.decorate('queryParser', viewDependencies.queryParser)
+  fastify.decorate('castResultsAsStream', viewDependencies.castResultsAsStream)
+  fastify.decorate('castItem', viewDependencies.castItem)
+  fastify.decorate('allFieldNames', viewDependencies.allFieldNames)
+  fastify.decorate('jsonSchemaGenerator', viewDependencies.jsonSchemaGenerator)
+  fastify.decorate('jsonSchemaGeneratorWithNested', viewDependencies.jsonSchemaGenerator)
+  fastify.decorate('modelName', modelName)
+  fastify.addHook('preHandler', (request, _reply, done) => {
+    for (const { as, localField } of lookups) {
+      if (request?.body?.[as]) {
```

Comment:
No, not in this case. This is because we want to fit each body to the underline collection interface, to be able to work with the view. The method check was only needed to return the update object with the correct interface, and this only happens in the PATCH of an id. The other calls return only the _id and, for that reason, it is not needed.